cpp/include/graph.hpp

```diff
@@ -214,5 +214,182 @@ class GraphCSC: public GraphCompressedSparseBase<VT,ET,WT> {
   {}
 };
 
+/**
+ * @brief       Base class for graphs constructed in the C++ API
+ *
+ * This class will own edge data, until the data is moved.
+ *
+ * @tparam VT   Type of vertex id
+ * @tparam ET   Type of edge id
+ * @tparam WT   Type of weight
+ */
+template <typename VT, typename ET, typename WT>
+class ConstructedGraphBase {
+public:
+  std::unique_ptr<WT>  edge_data{};     ///< edge weight
+  VT                   number_of_vertices{0};
+  ET                   number_of_edges{0};
+
+  ConstructedGraphBase() {}
+
+  ConstructedGraphBase(std::unique_ptr<WT> &&edge_data_, VT number_of_vertices_, ET number_of_edges_):
```

Comment:
Similarly, we should start a discussion on naming conventions for function input parameters for constructors and class member variables.

Here, you're adding _ at the of constructor input parameters. In other places, we are adding _ at the end (or at the beginning) of class member variables. And in yet other places, we are doing nothing. We need to do something and start to enforce consistency.

Reply:
Agreed

Reply:
I would vote for variables ending with underscore to be private variables.

Reply:
I agree with Aatish, that sounds more standard than adding trailing _ to function input parameters.